rust-runtime/aws-smithy-types/src/auth.rs

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+//! Smithy HTTP Auth Types
+
+use std::cmp::PartialEq;
+use std::fmt::Debug;
+
+#[derive(Debug, PartialEq)]
+enum AuthErrorKind {
```

Comment:
These types need to go into their own (new) `aws-smithy-http-auth` crate rather than into `aws-smithy-types`. I went ahead and squatted it so we can use the name: https://crates.io/crates/aws-smithy-http-auth

This new crate can be created in `rust-runtime/` side by side with `aws-smithy-types`.